org-formation/070-guard-duty/guardduty-s3-malware-protection.yaml

```diff
@@ -0,0 +1,147 @@
+       Version: 2012-10-17
+       Statement:
+          - Sid: AllowTrustToGuardDutyProtectionPlan
+            Effect: Allow
+            Principal:
+              Service: malware-protection-plan.guardduty.amazonaws.com
+            Action: 'sts:AssumeRole'
+  ManagedPolicy:
+    Type: "AWS::IAM::ManagedPolicy"
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: AllowManagedRuleToSendS3EventsToGuardDuty
+            Effect: Allow
+            Action:
+              - 'events:PutRule'
+              - 'events:DeleteRule'
+              - 'events:PutTargets'
+              - 'events:RemoveTargets'
+            Resource:
+              - !Sub 'arn:aws:events:${AWS::Region}:${AWS::AccountId}:rule/DO-NOT-DELETE-AmazonGuardDutyMalwareProtectionS3*'
+            Condition:
+              StringLike:
+                'events:ManagedBy': malware-protection-plan.guardduty.amazonaws.com
+          - Sid: AllowGuardDutyToMonitorEventBridgeManagedRule
+            Effect: Allow
+            Action:
+              - 'events:DescribeRule'
+              - 'events:ListTargetsByRule'
+            Resource:
+              - !Sub 'arn:aws:events:${AWS::Region}:${AWS::AccountId}:rule/DO-NOT-DELETE-AmazonGuardDutyMalwareProtectionS3*'
+          - Sid: AllowPostScanTag
+            Effect: Allow
+            Action:
+              - 's3:PutObjectTagging'
+              - 's3:GetObjectTagging'
+              - 's3:PutObjectVersionTagging'
+              - 's3:GetObjectVersionTagging'
+            Resource:
+              - !Sub 'arn:aws:s3:::${BucketName}/*'
+          - Sid: AllowEnableS3EventBridgeEvents
+            Effect: Allow
+            Action:
+              - 's3:PutBucketNotification'
+              - 's3:GetBucketNotification'
+            Resource:
+              - !Sub 'arn:aws:s3:::${BucketName}'
+          - Sid: AllowPutValidationObject
+            Effect: Allow
+            Action:
+              - 's3:PutObject'
+            Resource:
+              - !Sub 'arn:aws:s3:::${BucketName}/malware-protection-resource-validation-object'
+          - Sid: AllowCheckBucketOwnership
+            Effect: Allow
+            Action:
+              - 's3:ListBucket'
+            Resource:
+              - !Sub 'arn:aws:s3:::${BucketName}'
+          - Sid: AllowMalwareScan
+            Effect: Allow
+            Action:
+              - 's3:GetObject'
+              - 's3:GetObjectVersion'
+            Resource:
+              - !Sub 'arn:aws:s3:::${BucketName}/*'
+          - Sid: AllowDecryptForMalwareScan
+            Effect: Allow
+            Action:
+              - 'kms:GenerateDataKey'
+              - 'kms:Decrypt'
+            Resource: !Sub 'arn:aws:kms:${AWS::Region}:${AWS::AccountId}:key/*'   # bucket is encrypted using an AWS KMS key
+            Condition:
+              StringLike:
+                'kms:ViaService': s3.us-east-1.amazonaws.com
+  SnsTopic:
```

Comment:
I'm not sure if creating the topic in this stack is ideal, should the topic be a parameter instead? E.g. would setting up the topic subscriptions be a problem?

Reply:
@marcomarasca would passing in an existing SnsTopic and using that instead work for you?

Reply:
Sure I think that would work.

Reply:
Should we split the role from the plan (i.e. setup the account once such that multiple plans can be added later)?